mem/database.go

```diff
@@ -22,6 +22,6 @@ func (d *Database) Tables() map[string]sql.Table {
 	return d.tables
 }
 
-func (d *Database) AddTable(name string, t *Table) {
-	d.tables[name] = t
+func (d *Database) AddTable(t sql.Table) {
```

Comment:
Ok. Makes sense. If we want to change the name of a table we could probably add a wrapper `TableAlias` in the future.